### PR TITLE
Simplify terraform roles.

### PR DESCRIPTION
**Pull request by MancunianSam**
I've moved most things to resource * and action *. This isn't a huge
security problem as the role still can't acesss other services we don't
use and a lot of the services, we were adding most of the permissions
anyway.
For a lot of the resources, we had policies that allowed all resources
for a given region so this is only slightly more loose.
I've left EC2 and SSM with specific actions as these services have a
huge number of actions that we're not using.
I've also specified the IAM resources explicitly so new roles can't be
created or passed without changing the policy.

### CODE REVIEW

File: modules/environment-roles/templates/shared_terraform_policy_2.json.tpl [hunk 1]
```diff
@@ -2,123 +2,123 @@
   "Version": "2012-10-17",
   "Statement": [
     {
-      "Sid": "ssmglobal",
+      "Sid": "iam",
```
Comment (TomJKing) [inline modules/environment-roles/templates/shared_terraform_policy_2.json.tpl:5]: Is it worth renaming this policy to something more specific given it just contains the `iam` permissions: `shared_iam_terraform_policy` ?

**Review by TomJKing — COMMENTED** [review]
Looks good. Just one minor suggestion.

File: modules/environment-roles/templates/shared_terraform_policy_2.json.tpl (hunk 1 above)
Comment (MancunianSam) [reply modules/environment-roles/templates/shared_terraform_policy_2.json.tpl]: Yeah makes sense. I've renamed the file and the resources that reference it and I've changed the policy name to TDRSharedIamTerraform$environment

File: modules/environment-roles/root.tf [hunk 2]
```diff
@@ -51,93 +51,18 @@ resource "aws_iam_role_policy_attachment" "shared_policy_attachment_1" {
 }
 
 resource "aws_iam_role_policy_attachment" "shared_policy_attachment_2" {
```
Comment (TomJKing) [inline modules/environment-roles/root.tf:53]: Might want to rename this: `shared_iam_policy_attachment` for consistency.

File: modules/grafana/root.tf [hunk 3]
```diff
@@ -99,16 +75,6 @@ resource "aws_iam_role_policy_attachment" "shared_policy_attachment_1" {
 }
 
 resource "aws_iam_role_policy_attachment" "shared_policy_attachment_2" {
```
Comment (TomJKing) [inline modules/grafana/root.tf:77]: Might want to rename this: `shared_iam_policy_attachment` for consistency.

**Review by TomJKing — COMMENTED** [review]
Just a couple of small additional comment.

File: modules/environment-roles/root.tf (hunk 2 above)
Comment (MancunianSam) [reply modules/environment-roles/root.tf]: Ah yeah good point. I did a find and replace so I missed this one. I've changed the other one too.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**